spec/requests/preferences/preference_spec.rb

```diff
@@ -81,4 +81,88 @@
       end
     end
   end
+
+  describe 'GET edit' do
+    subject { get(edit_preference_path(preference.id)) }
+
+    let!(:user) { create(:user) }
+    let!(:preference) { create(:preference, user_id: user.id) }
+
+    context 'when logged in' do
+      before { sign_in user }
+
+      it 'have http status 200' do
+        expect(subject).to eq(200)
+      end
+    end
+
+    context 'when not logged in' do
+      it 'redirects to the sign-in page' do
+        subject
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  describe 'PUT update' do
+    subject { put preference_path(preference.id), params: }
+
+    let!(:user) { create(:user) }
+    let!(:preference) { create(:preference, user_id: user.id) }
+    let(:params) do
+      {
+        preference: {
+          name: 'DifferentName',
+          description: 'DifferentDescription',
+          restriction: false
+        }
+      }
+    end
+
+    context 'when logged in' do
+      before { sign_in user }
+
+      context 'when success' do
+        it 'have http status 302' do
+          expect(subject).to eq(302)
+        end
+
+        it 'redirect to index' do
+          expect(subject).to redirect_to(preferences_path)
+        end
+
+        it 'updates attributes correctly' do
+          subject
+          preference.reload
+          expect(preference.name).to eq(params.dig(:preference, :name))
+          expect(preference.description).to eq(params.dig(:preference, :description))
+          expect(preference.restriction).to eq(params.dig(:preference, :restriction))
```

Comment:
You can test exact values here. 
For example: 
`expect(preference.name).to eq('DifferentName')`

Reply:
Corrected!